Public/Src/Cache/ContentStore/Distributed/NuCache/TransitioningContentLocationStore.cs

```diff
@@ -249,6 +250,8 @@ public Task<BoolResult> UnregisterAsync(Context context, IReadOnlyList<ContentHa
         {
             Contract.Assert(_configuration.HasReadOrWriteMode(ContentLocationMode.LocalLocationStore), "GetLruPages can only be called when local location store is enabled");
 
+            context.Debug($"GetLruPages start with contentHashesWithInfo.Count={contentHashesWithInfo.Count}, firstAge={contentHashesWithInfo.FirstOrDefault().Age}, lastAge={contentHashesWithInfo.LastOrDefault().Age}");

```

Comment:
>LastOrDefault [](start = 196, length = 13)

LastOrDefault is not optimized for IReadOnlyList<T>. Maybe use contentHashesWithInfo[contentHashesWithInfo.Count - 1] instead. Maybe handle the empty collection separately.